tests/std/tests/P0475R1_P0591R4_uses_allocator_construction/test.cpp

```diff
@@ -179,6 +179,17 @@ void test_GH_2021() { // COMPILE-ONLY
     tags[0];
 }
 
+struct MoveOnlyType {
+    MoveOnlyType()               = default;
+    MoveOnlyType(MoveOnlyType&&) = default;
+};
+
+void test_LWG3527() { // COMPILE-ONLY
+    std::allocator<MoveOnlyType> alloc;
+    auto p                  = pair<MoveOnlyType&&, MoveOnlyType&&>{MoveOnlyType{}, MoveOnlyType{}};

```

Comment:
Style nitpick: Although we're more relaxed about test code than product code, we avoid Almost Always Auto (i.e. the pattern of initializing an `auto` variable from a temporary). Directly naming the type is simpler, and reserves `auto` for those cases where it is actually deducing a type from somewhere (as simple as a `static_cast` or something more complicated like a function).